### DISCUSSION TITLE
Homebrew installation

### DISCUSSION THREAD

**Discussion by Jay-Git168** [community/community#120063] [New to GitHub]
I try to install homebrew as instructed but my screen freezes and can't enter pw to proceed. pls help.
![2024-04-18_16-36-54](https://github.com/community/community/assets/76644466/01eb2393-e0ac-4d25-b0ad-b853aa441baa)

**Comment by ghost** [comment]
Hello Jay, 

From what I understand during the installation in the terminal, the password doesn't appear, correct?

This can happen for various reasons. If you're trying to type the password in the terminal and it's not showing up, it may be that the system is hiding the password input for security reasons. When you type the password in the terminal, it usually isn't displayed as visible characters, as a security measure to protect your password from being seen by other people who might be nearby.
If you're typing the password and not seeing anything on the screen, it's likely still being registered but not displayed. Try typing the password and pressing "Enter" to see if the process continues normally. If a password is being requested and you can't type anything, you may need to check if the keyboard is functioning correctly or if the input is being recognized by the system.

**Comment by Hadikamali** [comment]
Hi @Jay-Git168 
It looks like you are trying to install Homebrew on your MacBook Air, and the terminal is freezing when you need to enter your password for `sudo` access. Here are a few steps you can try to resolve this issue:

1. **Restart Terminal**:
   - Close the Terminal application completely and reopen it. Sometimes, a simple restart can resolve freezing issues.

2. **Run Terminal as Administrator**:
   - Ensure you are running Terminal with sufficient permissions. You can try using the `sudo` command to open a new terminal session with administrator privileges.

3. **Check for Updates**:
   - Ensure your macOS and Terminal application are up to date. Sometimes, updates include fixes for such issues.

4. **Safe Mode**:
   - Try booting your Mac in Safe Mode and then attempt the installation again. Safe Mode can sometimes help bypass certain software conflicts.

5. **Install Homebrew Manually**:
   - If the automated script is causing issues, you can try installing Homebrew manually by following the instructions on the [Homebrew website](https://brew.sh/).

6. **Check for Typographical Errors**:
   - Ensure there are no typographical errors in the command. The exact command should be:
     ```sh
     /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
     ```

7. **Permissions and Disk Space**:
   - Ensure you have sufficient disk space and permissions to install software on your Mac.

Here’s a step-by-step process to try:

1. Open Terminal.
2. Type the following command to install Homebrew:
   ```sh
   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
   ```
3. When prompted for a password, make sure to type it carefully. Note that the password won't be visible (not even asterisks) as you type, but it is being entered.
4. Press Enter after typing your password.

If the screen still freezes, try the following command to check for any active processes that might be causing the freeze:

```sh
ps aux | grep bash
```

You can terminate any suspicious processes using the `kill` command followed by the process ID (PID).

If none of these steps work, please provide more details about your macOS version and any error messages you receive.

**Comment by Jay-Git168** [comment]
Thanks. It works now.
Jay

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by Hadikamali** [reply]
Was my explanation helpful to you so that other people can fix this problem?
